Src/Support/GoogleApis.Auth.DotNet4/OAuth2/AuthorizationDialog.cs

```diff
@@ -0,0 +1,47 @@
+﻿using System;
+using System.Collections.Generic;
+using System.ComponentModel;
+using System.Data;
+using System.Drawing;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+using System.Windows.Forms;
+using Google.Apis.Auth.OAuth2.Responses;
+
+namespace Google.Apis.Auth.OAuth2
+{
+    public partial class AuthorizationDialog : Form
+    {
+        private AuthorizationCodeResponseUrl responseUrl;
+
+        public AuthorizationCodeResponseUrl ResponseUrl
+        {
+            get
+            {
+                return responseUrl;
+            }
+        }
+        /// <summary> Constructs a new authentication broker dialog. </summary>
+        public AuthorizationDialog(Uri authUri)
+        {
+            InitializeComponent();
+            // Set the browser to the initial uri.
+            authWebBrowser.Navigate(authUri);
+            // Need to track when the results has been received.
+            authWebBrowser.DocumentCompleted += authWebBrowser_DocumentCompleted;
```

Comment:
1. The DocumentCompleted event fires every time a URL is finished loading. For accept and deny, it will fire with a URL referencing localhost and the query portion of the response is assumed to be Google's AuthorizationCodeResponseUrl.
1. When the form is closed, the caller (EmbeddedBrowserCodeReceiver) has code that checks whether the response URL has been set. If the response has not been set, the task is set to the canceled state.
1. Something I have not tried is checking if the user gets a "localhost" response that is not a proper Google authorization response. This could happen, for example, if the user were to make the browser navigate to a localhost web server. I'm not certain if this is even possible since the embedded browser's navigation controls are disabled. Another potential problem is that a man-in-the-middle attack is sending the user to a malicious localhost. These would not be common scenarios, I hope.

In any case, I would expect that an exception will be thrown by the AuthorizationCodeResponseUrl constructor, which will be caught by the code receiver and associated with the task's result.